### ISSUE TITLE
CE polyfill is incompatible with babelHelpers class adapter

### ISSUE THREAD

**Issue by keanulee** [webcomponents/custom-elements#157]
The compiled CE definitions uses `babelHelpers` to make a class:

```js
  var DomModule =
  /*#__PURE__*/
  function (_HTMLElement) {
    babelHelpers.inherits(DomModule, _HTMLElement);

    function DomModule() {
      babelHelpers.classCallCheck(this, DomModule);
      return babelHelpers.possibleConstructorReturn(this, (DomModule.__proto__ || Object.getPrototypeOf(DomModule)).apply(this, arguments));
    }
```

The result of the constructor doesn't pass this if check:

https://github.com/webcomponents/custom-elements/blob/6374f375f35d05aae344ee3b61f1f7a4c0be2a81/src/CustomElementInternals.js#L266-L268

Which results in the `The custom element constructor did not produce the element being upgraded` error.

**Comment by web-padawan** [comment]
Related to Polymer/tools#182

**Comment by bicknellr** [comment]
It seems like this is a bug in babel: https://github.com/babel/babel/issues/7506 . Working on a fix here: https://github.com/Polymer/tools/pull/216 .

**Comment by bicknellr** [comment]
It's a strange bug though: they introduced a new behavior for what happens when you extend a built-in constructor that masks errors by preventing the super constructor from being called in the first place. They made something that wouldn't work with your ES2015 code work anyway work in the ES5 compiled version by doing something even further away from the native behavior. That bug and associated PR (https://github.com/babel/babel/pull/7533) sounds like they're reverting back to the 'try to call into the constructor, even if it would fail in pure ES5' behavior though.

**Comment by keanulee** [comment]
Addressed by https://github.com/Polymer/tools/pull/216, released in polymer-cli@1.7.0-pre.15
